src/node.js

```diff
@@ -22,6 +22,44 @@
 
     EventEmitter.call(process);
 
+    // Default process warning handler..
+    process.on('warning', (warning) => {
+      // By default, print the short warning message if --no-warnings is not
+      // set. If --trace-warnings is set, then the stack trace for the warning
+      // is shown. This takes backwards compatibility with --trace-deprecation
+      // into consideration also.
+      if (process.noProcessWarnings) return;
+      const trace = process.traceProcessWarnings ||
+                    (warning.name === 'DeprecationWarning' &&
+                     process.traceDeprecation);
+      if (trace)
+        console.error(warning.stack);
+      else
+        console.error(`${warning.name}: ${warning.message}`);
+    });
+
+    // process.emitWarning(error)
+    // process.emitWarning(str)
+    // process.emitWarning(options)
+    var Warning;
+    process.emitWarning = function(warning) {
```

Comment:
I actually *don't* want to override the name property of the `'Error'` object and instead just go ahead and pass that through. 

I allow `warning` to be an object to allow for the possibility that we may want to supply additional options at some point (see `depd`'s additional `namespace` property for instance). That may be a premature optimization, however. If so, switching this as you suggest may be appropriate.